src/util.ts

```diff
@@ -0,0 +1,85 @@
+import BN from "bn.js";
+import { Fraction } from "./fraction";
+
+const max128 = new BN(2).pow(new BN(128)).subn(1);
+
+/**
+ * Modifies the price to work with ERC20 units
+ * @param {number} price amount of quote token in exchange for one base token
+ * @param {number} baseTokenDecimals number of decimals of the base token
+ * @param {number} quoteTokenDecimals number of decimals of the quote token
+ * @return {Fraction} fraction representing the amount of units of quote tokens in exchange for one unit of base token
+ */
+export function getUnitPrice(
+  price: number,
+  baseTokenDecimals: number,
+  quoteTokenDecimals: number,
+): Fraction {
+  return Fraction.fromNumber(price).mul(
+    new Fraction(
+      new BN(10).pow(new BN(quoteTokenDecimals)),
+      new BN(10).pow(new BN(baseTokenDecimals)),
+    ),
+  );
+}
+
+/**
+ * Computes the amount of output token units from their price and the amount of input token units
+ * Note that the price is expressed in terms of tokens, while the amounts are in terms of token units
+ * @param {number} price amount of quote token in exchange for one base token
+ * @param {BN} baseTokenAmount amount of base token units that are exchanged at price
+ * @param {number} baseTokenDecimals number of decimals of the base token
+ * @param {number} quoteTokenDecimals number of decimals of the quote token
+ * @return {BN} amount of output token units obtained
+ */
+export function getOutputAmountFromPrice(
+  price: number,
+  baseTokenAmount: BN,
+  baseTokenDecimals: number,
+  quoteTokenDecimals: number,
+): BN {
+  const unitPriceFraction = getUnitPrice(
+    price,
+    baseTokenDecimals,
+    quoteTokenDecimals,
+  );
+  const quoteTokenAmountFraction = unitPriceFraction.mul(
+    new Fraction(baseTokenAmount, 1),
+  );
+  return quoteTokenAmountFraction.toBN();
+}
+
+/**
+ * Computes the quote and base token amounts needed to set up an unlimited order in the exchange
+ * @param {number} price amount of quote tokens in exchange for one base token
+ * @param {number} baseTokenDecimals number of decimals of the base token
+ * @param {number} quoteTokenDecimals number of decimals of the quote token
+ * @return {BN[2]} amounts of quote token and base token for an unlimited order at the input price
+ */
+export function getUnlimitedOrderAmounts(
+  price: number,
+  baseTokenDecimals: number,
+  quoteTokenDecimals: number,
+): [BN, BN] {
```

Comment:
I assume you just copied this, but how about changing the return type to a named struct so that there is no ambiguity which value is for quote and which value is for base token?

Reply:
I did not copy this, but If I understand correctly I could define a type called `SellAmount` and `BuyAmount` that are just BigNumbers so that people know exactly what is what.

Reply:
You could just use an anonymous named struct:

```suggestion
): { base: BN, quote: BN } {
```

You could define a `SellAmount` and `BuyAmount` type alias we well:
```
type SellAmount = BN;
type BuyAmount = BN;
```

And while that would document the function a bit better, it still wouldn't prevent callers doing this:
```
const [quoteAmount, baseAmount] = estimateBuyAmount(...);
```

While, using the anonymous named struct you get:
```
const { quote, base } = estimateBuyAmount(...);
// exactly the same as:
const { base, quote } = estimateBuyAmount(...);
```

Also, you can still easily set the names in the destructuring with:
```
const { quote: quoteAmount, base: baseAmount } = estimateBuyAmount(...);
// exactly the same as:
const { base: baseAmount, quote: quoteAmount } = estimateBuyAmount(...);
```

And even ignore one of the values, without having to worry about their order:
```
const { quote } = estimateBuyAmount(...);
// with a named parameter
const { quote: quoteAmount } = estimateBuyAmount(...);
```

Reply:
I have done the following.

``js
type SellAmount = BN;
type BuyAmount = BN;

export function getUnlimitedOrderAmounts(
  price: number,
  sellTokenDecimals: number,
  buyTokenDecimals: number,
): { base: SellAmount; quote: BuyAmount } 
```

Does this make sense?

Reply:
I guess my next question would be with regard to declaring types in the same file and then trying to import them later for use in unit tests. Seems my linter is unhappy.